cni/pkg/plugin/iptables.go

```diff
@@ -35,32 +38,28 @@ func newIPTables() InterceptRuleMgr {
 // Program defines a method which programs iptables based on the parameters
 // provided in Redirect.
 func (ipt *iptables) Program(netns string, rdrct *Redirect) error {
-	netnsArg := fmt.Sprintf("--net=%s", netns)
-	nsSetupExecutable := fmt.Sprintf("%s/%s", nsSetupBinDir, nsSetupProg)
-	nsenterArgs := []string{
-		netnsArg,
-		"--", // separate nsenter args from the rest with `--`, needed for hosts using BusyBox binaries
-		nsSetupExecutable,
-		"-p", rdrct.targetPort,
-		"-u", rdrct.noRedirectUID,
-		"-m", rdrct.redirectMode,
-		"-i", rdrct.includeIPCidrs,
-		"-b", rdrct.includePorts,
-		"-d", rdrct.excludeInboundPorts,
-		"-o", rdrct.excludeOutboundPorts,
-		"-x", rdrct.excludeIPCidrs,
-		"-k", rdrct.kubevirtInterfaces,
-	}
+	viper.Set(constants.CNIMode, true)
+	viper.Set(constants.NetworkNamespace, netns)
```

Comment:
is this safe to mutate? Seems like we are mutating global variables from within normal function. Does CNI call this multiple times? Or even worse, in parallel?

Reply:
Yeah, for every new pod, kubelet will exec a new CNI network plugin process. This function will only be executed once per process.